### PR TITLE
scalac: add page

### PR DESCRIPTION
**Pull request by ericchang00**
----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: pages/common/scalac.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# scalac
+
+> Scala compiler.
+
+- Compile a .scala file:
+
+`scalac {{filename.scala}}`
```
Comment (agnivade) [inline pages/common/scalac.md]: s/filename/file.

File: pages/common/scalac.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+- Compile several .scala files:
+
+`scalac {{filename1.scala}} {{filename2.scala}} {{filename3.scala}}`
+
+- Compile all .scala files in the current directory:
```
Comment (agnivade) [inline pages/common/scalac.md]: This example is a bit redundant.

**Comment by ericchang00** [comment]
@agnivade Thanks for the comments! I was keeping it consistent with the tldr page for [javac](https://github.com/tldr-pages/tldr/blob/master/pages/common/javac.md). Would you like to revise the `javac` page as well?

Regarding your comment on flags, most of them are infrequently used. `-classpath` may be worth adding.

**Comment by agnivade** [comment]
I see. Ok then, let us just do the filename => file change. If you think other options are not commonly used, no need to add them.

**Comment by ericchang00** [comment]
Sweet - done. The `javac` uses "filename" as well - might want to change that for consistency.

**Comment by agnivade** [comment]
> The javac uses "filename" as well - might want to change that for consistency.

Want to send a PR for that ? :wink:

**Comment by ericchang00** [comment]
Haha yep 👍

**Comment by pxgamer** [comment]
@sbrl, looks like it's `-d {{path/to/directory}}` or `-d {{path/to/file.jar}}` for choosing the output path.

**Comment by agnivade** [comment]
@ericchang00  - Do you not use the `-d` option usually ?

**Comment by ericchang00** [comment]
Honestly, not really. `sbt` is the standard build tool for any sizable project. Typically I only use `scala` for one off scripts.

**Comment by sbrl** [comment]
Hrm, I see. Normally with a compiler I'd suggest including it - even if it's just part of a single example - but it's up to you (I'm not a scala programmer :P)

**Pull request closed without merging**